### PR TITLE
React, Part 3

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: content/react-lsn3/reading/state-intro/index.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
...me money, the number is red. At the beginning of the process, both federal and state would have shown $0 or "N/A" to indicate that you haven't started the process yet. Now when you enter the information on a form, like your W2, you see the numbers in that component changing. In fact, every time you edit some aspect of your taxes, the numbers may change and the colors of that number may change. This is where state comes into play in this React-based tax application. Reloading the page every time you edit one thing would be annoying and make the application inefficient. Even just re-rendering one component on the page can be a costly process. With state, the component is only re-rendered when there is new data. If you enter some information that doesn't change the fact that the federal government owes you a refund of $200, the component will not re-render. Selectively re-rendering components makes React an efficient library to use for web development. While not always noticeable, re-rendering components can lead to lagging or delays as the users interacts with your application. By only re-rendering components in select cases, React helps us minimize the possibility of this happening.
```
Comment (codinglikeagirl42) [inline content/react-lsn3/reading/state-intro/index.md:17]:
```suggestion
Let's think back to the last time you filled out your tax return. If you e-file your taxes, you might have used a React-based web application to do so. An example component would be the tax estimate for your federal and state taxes. For some tax applications, if it looks like you are getting a refund, the number is green and if it looks like you owe some money, the number is red. At the beginning of the process, both federal and state would have shown $0 or "N/A" to indicate that you haven't started the process yet. Now when you enter the information on a form, like your W2, you see the numbers in that component changing. In fact, every time you edit some aspect of your taxes, the numbers may change and the colors of that number may change. This is where state comes into play in this React-based tax application. Reloading the page every time you edit one thing would be annoying and make the application inefficient. Even just re-rendering one component on the page can be a costly process. With state, the component is only re-rendered when there is new data. If you enter some information that doesn't change the fact that the federal government owes you a refund of $200, the component will not re-render. Selectively re-rendering components makes React an efficient library to use for web development. While not always noticeable, re-rendering components can lead to lagging or delays as the users interact with your application. By only re-rendering components in select cases, React helps us minimize the possibility of this happening.
```

File: content/react-lsn3/reading/state-intro/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+While state is a built-in concept in React, you as the programmmer have to call upon it. Local variables won't keep their values during re-renders and changing the value of a local variable doesn't trigger re-rendering. We have to use a **state variable** to hold data that we want to keep as part of the component's memory. To set the value of a state variable, we will use a **state setter function**. Both a state variable and state setter function are provided by a React hook called `useState()`. **Hooks** are features of React that allow you to access different aspects of React and use them with your components. Without hooks, components would just be ordinary classes and functions.
```
Comment (codinglikeagirl42) [inline content/react-lsn3/reading/state-intro/index.md:19]:
```suggestion
While state is a built-in concept in React, you as the programmer have to call upon it. Local variables won't keep their values during re-renders and changing the value of a local variable doesn't trigger re-rendering. We have to use a **state variable** to hold data that we want to keep as part of the component's memory. To set the value of a state variable, we will use a **state setter function**. Both a state variable and state setter function are provided by a React hook called `useState()`. **Hooks** are features of React that allow you to access different aspects of React and use them with your components. Without hooks, components would just be ordinary classes and functions.
```

File: content/react-lsn3/reading/state-intro/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+In the example of a tax application, we might have a state variable for federal taxes called `federalTax`. We can then declare that state variable like so:
+
+```jsx
+   const [federalTax, setFederalTax] = useState(0);
+```
+
+Here we are using the `useState` hook to set the initial value of `federalTax` to 0. We also have declared that the state setter function for `federalTax` is `setFederalTax`. To update the value of `federalTax` to 100, we just have to use `setFederalTax` like so, `setFederalTax(100)`. 
+
+You can also set the up the state variable to be an object. Here is how we might do it with our tax application.
```
Comment (codinglikeagirl42) [inline content/react-lsn3/reading/state-intro/index.md:29]:
```suggestion
You can also set up the state variable to be an object. Here is how we might do it with our tax application.
```

File: content/react-lsn3/reading/form-intro/index.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                  </label>
+                  <input type="submit" />
+               </form>
+
+               <p>Preview: {notes}</p>
+            </div>);
+      }
+   ```
+
+1. Now when a user types text into the form field, the state of the component is continually changed and the words the user types appears in the `<p>` element.
```
Comment (codinglikeagirl42) [inline content/react-lsn3/reading/form-intro/index.md:86]:
```suggestion
1. Now when a user types text into the form field, the state of the component is continually changed and the words the user types appear in the `<p>` element. 
```

File: content/react-lsn3/exercises/_index.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+Time to make a React app to share a hobby you love with your friends and family!
+
+{{% notice blue "Note" "rocket" %}}
+
+The solution in the `solutions` branch of the `react-exercises-and-studios` repo uses needlepoint as an example hobby. Whenever you see a reference to needlepoint in the solution to the exercises, subsitute it for your own chosen hobby.
+
+{{% /notice %}}
+
+1. Create a new application called `exercises` in the `part3` directory in the `react-exercises-and-studios` repo.
+1. In `App.js`, remove the content from the `<div className="App">` element and replace it with an `h1` heading that says "My Hobby: " with your hobby coming after the colon.
```
Comment (codinglikeagirl42) [inline content/react-lsn3/exercises/_index.md:24]:
```suggestion
1. In `App.js`, remove the content from the `<div className="App">` element and replace it with an `<h1>` heading that says "My Hobby: " with your hobby coming after the colon.
```

File: content/react-lsn3/exercises/_index.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+   ```jsx
+   export default function HobbyIntroduction () {
+      return();
+   }
+   ```
+   {{% /expand %}}
+
+1. This functional component return a `<div>` that contains the following:
+
+   1. An `<h2>` heading that that says something to the effect of "3 Fun Facts About this Hobby".
```
Comment (codinglikeagirl42) [inline content/react-lsn3/exercises/_index.md:47]:
```suggestion
   1. An `<h2>` heading that says something to the effect of "3 Fun Facts About this Hobby".
```

File: content/react-lsn3/studio/_index.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+Let's add some additional functionality to our recipe app! To get started, check out the starter code for the studio in the `part3` directory on the `main` branch of `react-exercises-and-studios`. The starter code includes all the components you will need and in `App.js`, all the components are laid out for you. However, if you run the code, you will find lots of warnings and errors. Time to check out the different components and make use of what you have learned about state!
```
Comment (codinglikeagirl42) [inline content/react-lsn3/studio/_index.md:15]: Suggestion to add links to the GitHub Repo.

File: content/react-lsn3/exercises/_index.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+Time to make a React app to share a hobby you love with your friends and family!
+
+{{% notice blue "Note" "rocket" %}}
+
+The solution in the `solutions` branch of the `react-exercises-and-studios` repo uses needlepoint as an example hobby. Whenever you see a reference to needlepoint in the solution to the exercises, subsitute it for your own chosen hobby.
```
[the remark this answers is not in the document]
Comment (codinglikeagirl42) [reply content/react-lsn3/exercises/_index.md:19]: Also suggest adding a link to the GitHub Repo.

**Review by codinglikeagirl42 — APPROVED** [review]
A few small suggestions, then merge away!

File: content/react-lsn3/studio/_index.md (hunk 7 above)
Comment (gildedgardenia) [reply content/react-lsn3/studio/_index.md:15]: Done!

**Pull request merged**